src/main/java/gov/usds/case_issues/db/repositories/TroubleCaseRepository.java

```diff
@@ -40,6 +44,11 @@
 	@Query(ACTIVE_CASE_QUERY + " and " + ACTIVE_SNOOZE_CLAUSE)
 	public Page<TroubleCase> getSnoozedWithOpenIssues(CaseManagementSystem caseManagementSystem, CaseType caseType, Pageable pageable);
 
+	@Lock(LockModeType.PESSIMISTIC_WRITE) // might need to be more aggressive when postgresql table-level LOCK is available
```

Comment:
Eventual solution, where eventually is probably the same time that we make these things asynchronous: https://www.postgresql.org/docs/9.0/sql-lock.html